instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -290,15 +338,37 @@ def collect_custom_request_headers_attributes(scope):
     Refer specification https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers"""
 
     attributes = {}
-    custom_request_headers = get_custom_headers(
+
+    sanitized_fields = get_custom_headers(
+        OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS
+    )
+
+    sanitize = SanitizeValue(sanitized_fields)
+
+    custom_request_headers_name = get_custom_headers(
         OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST
     )
 
-    for header in custom_request_headers:
-        values = asgi_getter.get(scope, header)
-        if values:
-            key = normalise_request_header_name(header)
-            attributes.setdefault(key, []).extend(values)
+    if custom_request_headers_name:
+        custom_request_headers_regex_compiled = re.compile(
+            "|".join("^" + i + "$" for i in custom_request_headers_name),
+            re.IGNORECASE,
+        )
+
+        for header_name in list(
+            filter(
+                custom_request_headers_regex_compiled.match,
+                asgi_getter.keys(scope),
+            )
+        ):
+            header_values = asgi_getter.get(scope, header_name.lower())
+            if header_values:
+                key = normalise_request_header_name(header_name.lower())
+                attributes[key] = [
+                    sanitize.sanitize_header_value(
+                        header=header_name, value=header_values[0]
+                    )
+                ]
```

Comment:
Could you add this to util-http package so that it will be easier to use in other instrumentation?

Reply:
Done.